### PR TITLE
Add support for Python 3.9 build testing

### PR DESCRIPTION
**Pull request by micahellison**
First step of #1017 - add build steps for Python 3.9 in Travis. Allowing failures for now, until 3.9 is stable.

### Checklist

- [X] I have read the [contributing doc](https://github.com/jrnl-org/jrnl/blob/develop/CONTRIBUTING.md).
- [X] I have included a link to the relevant issue number.
- [n/a] I have tested this code locally.
- [X] I have checked to ensure there aren't other open [pull requests](../pulls)
  for the same issue.
- [n/a] I have written new tests for these changes, as needed.
- [n/a] All tests pass.

### CODE REVIEW

**Comment by micahellison** [comment]
This still needs some work. Issues to figure out:

* `before_install` step
  * Failing on 3.9 Mac. Looks like the `sed` command behaves differently on MacOS, and I'm not sure how best to replace it with find-and-replace logic.
  * Failing on 3.9 Windows. It works fine on my local machine but maybe a different version of PowerShell is running on Travis.
  * Maybe we could do this more effectively with a Python command executed from the shell, and use the same command for all OSes.
* I have somehow managed to add the 3.9 Linux build to `allow_failures` but not the others. I don't know why. I want all of them to be allowed to fail for now.
* Some new test failures on the 3.9 Linux build.
  * Is `behave` just dying in the middle of tests?
  * Tests are failing due to the version string: "ERROR: Python version 3.9.0b5+ not supported." Is it the case that any beta Python build is not allowed to run jrnl now? And is that okay?

To anyone looking at this: feel free to offer thoughts and suggestions. I'll come back to this later this week.

**Comment by wren** [comment]
Yeah, the sed command was a hack thrown in there when we didn't have to consider the other OSes. We should probably use the toml library we already load in our python dev deps.

**Comment by micahellison** [comment]
I've resolved the `allow_failures` problem, and I've got it working on Mac and Linux (thanks for the help @wren) but I'm still struggling with Windows. It's installing 3.9.0b5 with `choco install python --pre` but I don't know where it's installing it, since there's also a Python 2.7 installation on this Windows machine, and Python isn't behaving quite like it does in the other Windows Python 3 builds.

I'll come back to this later but, as always, any help is appreciated.

**Comment by micahellison** [comment]
Okay. I think this is about as good as I'm going to get this.

Windows is correctly using Python 3.9, but it can't install cryptography because there are no wheels available for Python 3.9. The alternative is building it from source, which feels like a bit much, and isn't very representative of Windows users anyway, as they are generally not going to take this approach.. Maybe it will work as we get closer to the release in October, or if not, we'll at least know about this issue when we cross that bridge.

File: .travis.yml [hunk 1]
```diff
@@ -105,10 +108,42 @@ jobs:
...
+        - python .build/allow_all_python_version.py
+      env:
+        <<: *env_windows
+        JRNL_PYTHON_VERSION: 3.9.0b5
+
     # ... and beyond!
     - name: Python nightly on Linux
       before_install:
-        - sed -i 's/^python = ">=3\.6\.0.*"$/python = "*"/' pyproject.toml
+        - sed -i 's/^python = ">=3\.7\.0.*"$/python = "*"/' pyproject.toml
```
Comment (wren) [inline .travis.yml]: Shouldn't this use your new script (` .build/allow_all_python_version.py`)?

**Review by wren — DISMISSED** [review]
Looks great with one minor comment. I'll be happy to merge if you don't want to fix it, just let me know.

**Review by wren — APPROVED** [review]
🐆

**Pull request merged**